### PR TITLE
Update HeadSpin Panel

### PR DESCRIPTION
**Pull request by KazuCocoa**
- after

- before

### CODE REVIEW

File: app/renderer/actions/Session.js [hunk 1]
```diff
@@ -210,12 +211,24 @@ export function newSession (caps, attachSessId = null) {
...
-        host = session.server.headspin.hostname;
-        port = session.server.headspin.port;
-        path = `/v0/${session.server.headspin.apiKey}/wd/hub`;
-        https = true;
+      case ServerTypes.headspin: {
+        if (!/^(http|https):\/\/(localhost|.+\.headspin.io)(|:)[0-9]*\/v\d+\/\w+\/wd\/hub$/.test(
+          session.server.headspin.webDriverUrl)) {
+          notification.error({
+            message: i18n.t('Error'),
+            description: i18n.t('invalid HeadSpin Web Driver URL. Make sure it'),
```
Comment (mykola-mokhnach) [inline app/renderer/actions/Session.js]: Make sure it ... what?

File: app/renderer/actions/Session.js [hunk 2]
```diff
@@ -210,12 +211,24 @@ export function newSession (caps, attachSessId = null) {
...
+            duration: 4
+          });
+          return;
+        }
+        const headspinUrl = url.parse(session.server.headspin.webDriverUrl);
+
+        host = headspinUrl.hostname;
+        port = headspinUrl.port;
+        path = headspinUrl.pathname;
+        https = headspinUrl.protocol === 'https:' ? true : false;
```
Comment (mykola-mokhnach) [inline app/renderer/actions/Session.js]: `https = headspinUrl.protocol === 'https:'` is enough

File: app/renderer/actions/Session.js [hunk 3]
```diff
@@ -210,12 +211,24 @@ export function newSession (caps, attachSessId = null) {
           desiredCapabilities.testobject_api_key = session.server.testobject.apiKey || process.env.TESTOBJECT_API_KEY;
         }
         break;
-      case ServerTypes.headspin:
-        host = session.server.headspin.hostname;
-        port = session.server.headspin.port;
-        path = `/v0/${session.server.headspin.apiKey}/wd/hub`;
-        https = true;
+      case ServerTypes.headspin: {
+        if (!/^(http|https):\/\/(localhost|.+\.headspin.io)(|:)[0-9]*\/v\d+\/\w+\/wd\/hub$/.test(
```
Comment (mykola-mokhnach) [inline app/renderer/actions/Session.js]: regexp matching is potentially error-prone. Why not to try url parsing first and then verify the parsed properties?

File: app/renderer/actions/Session.js [hunk 4]
```diff
@@ -212,21 +212,20 @@ export function newSession (caps, attachSessId = null) {
...
         break;
       case ServerTypes.headspin: {
-        if (!/^(http|https):\/\/(localhost|.+\.headspin.io)(|:)[0-9]*\/v\d+\/\w+\/wd\/hub$/.test(
-          session.server.headspin.webDriverUrl)) {
+        const headspinUrl = url.parse(session.server.headspin.webDriverUrl);
+        host = headspinUrl.hostname;
+        port = headspinUrl.port;
+        path = headspinUrl.pathname;
+        https = headspinUrl.protocol === 'https:';
+        if (!/^(localhost|.+\.headspin.io)$/.test(host)
```
Comment (mykola-mokhnach) [inline app/renderer/actions/Session.js]: I assume dots should be properly escaped in the regexp

File: app/renderer/actions/Session.js [hunk 5]
```diff
@@ -212,21 +212,20 @@ export function newSession (caps, attachSessId = null) {
...
+          || !/v\d+\/\w+\/wd\/hub$/.test(path)) {
           notification.error({
             message: i18n.t('Error'),
-            description: i18n.t('invalid HeadSpin Web Driver URL. Make sure it'),
+            description: i18n.t('invalid HeadSpin Web Driver URL. Please make sure the Web Driver URL'),
```
Comment (mykola-mokhnach) [inline app/renderer/actions/Session.js]: Dots are not allowed in i18next identifiers. Please keep the unique id here and put the translatable string in locales/en/translation.json

Comment (KazuCocoa) [reply app/renderer/actions/Session.js]: ah.. i see

File: assets/locales/en/translation.json [hunk 6]
```diff
@@ -215,9 +215,8 @@
   "Perfecto Token": "Perfecto Token",
   "Your Kobiton Username": "Your Kobiton Username",
   "Kobiton Access Key": "Kobiton Access Key",
-  "Headspin Host": "Headspin Host",
-  "Headspin Port": "Headspin Port",
-  "Headspin API Token": "Headspin API Token",
+  "serverTabHeasdpinWebDriverURL": "HeadSpin Web Driver URL",
+  "sessionHeadspinInvalidWebDriverURLError": "invalid HeadSpin Web Driver URL. Please make sure the Web Driver URL",
```
Comment (mykola-mokhnach) [inline assets/locales/en/translation.json]: The HeadSpin WebDriver URL is not valid. Please enter the correct URL it and try again.

**Pull request merged**